Confusing error now saying `let` (not `if let`) is an expression, rather than a statement, which contradicts the book

cc @Centril 

Code first. This code:

```
fn main() {
    let x = (let y = 6);
}
```

[on current stable Rust 1.38](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=db087215a59f29b6c19440c89ddb2484), gives this set of error messages:

```
error[E0658]: `let` expressions in this position are experimental
 --> src/main.rs:2:14
  |
2 |     let x = (let y = 6);
  |              ^^^^^^^^^
  |
  = note: for more information, see https://github.com/rust-lang/rust/issues/53667

error: `let` expressions are not supported here
 --> src/main.rs:2:14
  |
2 |     let x = (let y = 6);
  |              ^^^^^^^^^
  |
  = note: only supported directly in conditions of `if`- and `while`-expressions
  = note: as well as when nested within `&&` and parenthesis in those conditions

warning: unnecessary parentheses around assigned value
 --> src/main.rs:2:13
  |
2 |     let x = (let y = 6);
  |             ^^^^^^^^^^^ help: remove these parentheses
  |
  = note: `#[warn(unused_parens)]` on by default
``` 

The part that I'm most concerned about is:

```
error: `let` expressions are not supported here
```

Before eRFC *"if- and while-let-chains, take 2"* (rust-lang/rfcs#2497, https://github.com/rust-lang/rust/issues/53667, https://github.com/rust-lang/rust/issues/53668), this code USED to result in this error message:

```
error: expected expression, found statement (`let`)
 --> src/main.rs:2:14
  |
2 |     let x = (let y = 6);
  |              ^^^
  |
  = note: variable declaration using `let` is a statement
```

The reason I know this is because [we have the old error message in the book](https://doc.rust-lang.org/1.38.0/book/ch03-03-how-functions-work.html#function-bodies-contain-statements-and-expressions), in the section where we're trying to explain the difference between statements and expressions. The error message I'm seeing now is muddying the waters by saying "`let` expressions".

Based on my reading of the eRFC, it's only supposed to change `if let`, but as kind of a side effect `let` is now sort-of an expression? The [updates to the reference](https://github.com/rust-lang-nursery/reference/pull/437/files) don't clear it up for me, as they *only* describe `if let`, not plain `let`s.

What I expected is that even though the eRFC has been accepted and implemented, plain `let y = 6` would still be considered a statement and the error message wouldn't talk about "`let` expressions".

If my expectation is valid, then the compiler error message is a bug. If my expectation is invalid, please let me know so that I can work on updating the book. Thanks!

<!-- TRIAGEBOT_START -->

<!-- TRIAGEBOT_ASSIGN_START -->
This issue has been assigned to @jafern14 via [this comment](https://github.com/rust-lang/rust/issues/65254#issuecomment-546919992).
<!-- TRIAGEBOT_ASSIGN_DATA_START$${"user":"jafern14"}$$TRIAGEBOT_ASSIGN_DATA_END -->

<!-- TRIAGEBOT_ASSIGN_END -->
<!-- TRIAGEBOT_END -->

--- a/src/librustc/hir/lowering/expr.rs
+++ b/src/librustc/hir/lowering/expr.rs
@@ -235,11 +235,20 @@ impl LoweringContext<'_> {
     /// ```
     fn lower_expr_let(&mut self, span: Span, pat: &Pat, scrutinee: &Expr) -> hir::ExprKind {
         // If we got here, the `let` expression is not allowed.
-        self.sess
-            .struct_span_err(span, "`let` expressions are not supported here")
-            .note("only supported directly in conditions of `if`- and `while`-expressions")
-            .note("as well as when nested within `&&` and parenthesis in those conditions")
-            .emit();
+
+        if self.sess.opts.unstable_features.is_nightly_build() {
+            self.sess
+                .struct_span_err(span, "`let` expressions are not supported here")
+                .note("only supported directly in conditions of `if`- and `while`-expressions")
+                .note("as well as when nested within `&&` and parenthesis in those conditions")
+                .emit();
+        }
+        else {
+            self.sess
+                .struct_span_err(span, "expected expression, found statement (`let`)")
+                .note("variable declaration using `let` is a statement")
+                .emit();
+        }
 
         // For better recovery, we emit:
         // ```

